h2quic/request.go

```diff
@@ -37,7 +37,7 @@ func requestFromHeaders(headers []hpack.HeaderField) (*http.Request, error) {
 		httpHeaders.Set("Cookie", strings.Join(httpHeaders["Cookie"], "; "))
 	}
 
-	if len(path) == 0 || len(authority) == 0 || len(method) == 0 {
+	if (len(path) == 0 && method != http.MethodConnect) || len(authority) == 0 || len(method) == 0 {
```

Comment:
This looks fine, this is what the H2 standard specifies (https://http2.github.io/http2-spec/index.html#CONNECT).

There's one thing I'd like to check before approving this PR. We're parsing the path into an URL a couple of lines later here. Parsing an empty string works fine with `url.Parse`, however, I'm wondering if the returned `http.Request` should have a `nil` value for `http.Request.URL`, or this empty URL. Could you setup an example with a Go H2 server to check how the standard library handles that (please paste the code here or a link to a Go playground file)?